src/primitive_esp32.cc

```diff
@@ -662,6 +690,76 @@ PRIMITIVE(pm_get_configuration) {
 #endif
 }
 
+PRIMITIVE(pm_lock_new) {
+  ARGS(SimpleResourceGroup, group, int, lock_type_value, cstring, name);
+
+  esp_pm_lock_type_t lock_type;
+  switch (lock_type_value) {
+    case 0: lock_type = ESP_PM_CPU_FREQ_MAX; break;
+    case 1: lock_type = ESP_PM_APB_FREQ_MAX; break;
+    case 2: lock_type = ESP_PM_NO_LIGHT_SLEEP; break;
+    default: FAIL(INVALID_ARGUMENT);
+  }
+
+  ByteArray* proxy = process->object_heap()->allocate_proxy();
+  if (proxy == null) FAIL(ALLOCATION_FAILED);
+
+  bool handed_to_resource = false;
+
+  char* copied_name = strdup(name);
+  if (copied_name == null) FAIL(MALLOC_FAILED);
+  Defer del_string { [&] { if (!handed_to_resource) free(copied_name); } };
+
+  esp_pm_lock_handle_t handle;
+  esp_err_t err = esp_pm_lock_create(lock_type, 0, copied_name, &handle);
+  if (err != ESP_OK) return Primitive::os_error(err, process);
+  Defer release_lock { [&] { if (!handed_to_resource) esp_pm_lock_release(handle); } };
+
+  auto resource = _new PmLockResource(group, handle, copied_name);
+  if (resource == null) FAIL(ALLOCATION_FAILED);
+  handed_to_resource = true;
```

Comment:
This isn't super clean, so maybe we need a "public" way of doing this, but it is kinda silly to do all this copying of the C strings.

```suggestion
  handed_to_resource = true;
  _manager_name.keep_result();
```